client/dashboard/app/analytics/tracking-preferences.ts

```diff
@@ -0,0 +1,216 @@
+import cookie from 'cookie';
+
+type TrackingPrefs = {
+	ok: boolean;
+	buckets: {
+		essential: boolean;
+		analytics: boolean;
+		advertising: boolean;
+	};
+};
+
+type TrackingPrefsData = Partial<
+	Omit< TrackingPrefs, 'buckets' > & { buckets: Partial< TrackingPrefs[ 'buckets' ] > }
+>;
+
+const COOKIE_MAX_AGE = 60 * 60 * 24 * ( 365.25 / 2 ); // 365.25 is the average number of days in a year.
+const TRACKING_PREFS_COOKIE_V1 = 'sensitive_pixel_option';
+const TRACKING_PREFS_COOKIE_V2 = 'sensitive_pixel_options';
+
+const prefsDisallowAll: TrackingPrefs = {
+	ok: false,
+	buckets: {
+		essential: true, // essential bucket is always allowed
+		analytics: false,
+		advertising: false,
+	},
+};
+
+const prefsAllowAnalyticsGdpr: TrackingPrefs = {
+	ok: false, // false is important so the cookie banner is shown
+	buckets: {
+		essential: true,
+		analytics: true, // in GDPR zone, analytics is opt-out
+		advertising: false, // in GDPR zone, advertising is opt-in
+	},
+};
+
+const prefsAllowAll: TrackingPrefs = {
+	ok: true,
+	buckets: {
+		essential: true,
+		analytics: true,
+		advertising: true,
+	},
+};
+
+/**
+ * Returns a boolean telling whether a country is in the GDPR zone.
+ * @param countryCode The country code to look for.
+ * @returns Whether the country is in the GDPR zone
+ */
+export function isCountryInGdprZone( countryCode: string | undefined ): boolean {
```

Comment:
Should we update the existing code to use this function?

Otherwise I'm not sure how to be confident that when we release v2, the country list is still up to date. 😄

Reply:
Good idea, I'd prefer to address this in a separate PR so that the changes are self-containing.